activesupport/lib/active_support/values/time_zone.rb

```diff
@@ -435,9 +435,18 @@ def iso8601(str)
     #
     #   Time.zone.parse('Mar 2000') # => Wed, 01 Mar 2000 00:00:00 HST -10:00
     #
+    # As with +Time.parse+, a block can be provided to control how year values are handled:
+    #
+    #   Time.zone.parse("70-10-31") { |year| year + (year < 70 ? 2000 : 1900) }
+    #   # => Sat, 31 Oct 1970 00:00:00.000000000 HST -10:00
+    #
     # If the string is invalid then an +ArgumentError+ could be raised.
     def parse(str, now = now())
-      parts_to_time(Date._parse(str, false), now)
+      comp = !block_given?
+      parts = Date._parse(str, comp)
+      parts[:year] = yield(parts[:year]) if parts[:year] && !comp
+
+      parts_to_time(parts, now)
```

Comment:
Do we need to change the `comp` value on whether a block is given? Does the following not work the same?

``` ruby
def parse(str, now = now())
  parts = Date._parse(str, false)
  parts[:year] = yield(parts[:year]) if parts[:year] && block_given?

  parts_to_time(parts, now)
end
```

Reply:
Sorry for the delay getting back to this @pixeltrix!

I'd been trying to stay as close as possible to [Ruby's definition of `Time.parse`](https://github.com/ruby/ruby/blob/203eeeefddb3ae6c0e129ad9cd99da804c2cbaba/lib/time.rb#L378-L382), which notably passes a _truthy_ second argument to `Date._parse` if a block isn't given.

I've been trying to work out whether there is any other behaviour difference with the `comp` value but have been finding the [underlying definition](https://github.com/ruby/ruby/blob/master/ext/date/date_parse.c#L2090) harder to grok. I think it is just controlling the value of `parts[:year]` that is getting yielded to the block.

I was ultimately just shooting for parity in behaviour between `Time.parse` and `Time.zone.parse` via parity in implementation! 😳